addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1281,11 +1302,27 @@ const InputUserValueWidget = UnitUserValueWidget.extend({
                 }
                 value += (ev.which === $.ui.keyCode.UP ? step : -step);
                 input.value = this._floatToStr(value);
+                // We need to know if the change event will be triggered or not.
+                // Change is triggered if there has been a "natural" input event
+                // from the user. Since we are triggering a "fake" input event,
+                // it should not change the changeEventWillBeTriggered variable.
+                const changeEventWillBeTriggered = this.changeEventWillBeTriggered;
                 $(input).trigger('input');
+                this.changeEventWillBeTriggered = changeEventWillBeTriggered;
+                this.notifyValueChangeOnBlur = true;
```

Comment:
We can actually use a custom event to give more details to the _onInputInput handler:

in `_onInputKeydown` instead of `$(input).trigger('input');`
```js
        input.dispatchEvent(new CustomEvent('input', {
            bubbles: true,
            cancelable: true,
            detail: {keyUpOrDown: true}
        }));
```
then in `_onInputInput`
```js
    _onInputInput: function (ev, ) {
        ...
        if (!ev.detail || !ev.detail.keyUpOrDown) {
            this.changeEventWillBeTriggered = true;
        }
        ...
    },
```

This would remove the need to save the state prior to the event triggering, which I'm not a huge fan of as it could break if an event handler becomes async (I don't think the case exists in the code base and maybe it should never exist in the first place but still).

I did a quick test and it seems to work, but let me know what you think about it 😄